### ISSUE TITLE
Randomly fails to update contents of chat - e.g. message deleted by bot remains visible in Telegram Desktop

### ISSUE THREAD

**Issue by php4fan** [telegramdesktop/tdesktop#27353]
### Steps to reproduce

Obviously there's no way I'll ever be able to provide steps to reproduce.

This is what happens:

1. In a supergroup, some user sends a message to the chat
2. a bot that is admin of the chat, deletes the user's message almost immediately

### Expected behaviour

The deleted message should disappear in Telegram Desktop.
One may expect a delay of a few seconds at most.

### Actual behaviour

Usually messages that are deleted by the bot (or by anybody for that matter) disappear as expected, but sometimes, randomly:
The message remains visible and never gets deleted in Telegram Desktop, unless I restart the application.

Messages that have been sent after the one that was deleted, are visible: so it's not like the connection has been lost or anything. The connection is working, the chat is syncing, but some particular update (i.e. the deletion of a message in particular) has been silently lost and never gets synced. And there's no indication anywhere that something is wrong. The user has every right to assume that what they are seeing is the current up-to-date contents of the chat, but that's not the case.

Obviously I checked in the mobile app in the same group and the message was deleted.

### Operating system

Manjaro Linux

### Version of Telegram Desktop

4.12.2

### Installation source

Static binary from official website

**Comment by php4fan** [comment]
This keeps happening

**Comment by php4fan** [comment]
Again the stupid bot, attempting to close issues "because" no progress have been made on them.

**Comment by php4fan** [comment]
Why the fuck is this closed??

This still keeps happening

**Comment by php4fan** [comment]
This MIGHT actually be an issue in Telegram itself.  If so, you guys should report this to your colleagues  at the Backend/API side. This is VERY serious, this can have privacy repercussions.

**Comment by 23rd** [comment]
Any available public example?

**Comment by php4fan** [comment]
Thanks for reopening.

Nope, no public example available. We have a couple dozen groups managed by this bot which are all private groups. Additionally, every time this happens, either I or someone else manually delete the messages that had been deleted by the group and were still visible (at least when they are still visible to us), which fortunately seems to work (unlike any attempt by the bot to delete them again, which results in an error response saying the message doesn't exist). It's actually kind of wild that that works (and doesn't give any sort of error message or quirk, just as if the message was still there and as if the second, manual deletion actually deleted it), but that's a good thing for us. We obviously do that for the same reasons the bot had deleted the messages in the first place. 

So, any message that was affected is no longer available (and we cannot leave one for you to see the next time it happens, because these are messages that are deleted for good reasons and leaving them there would put us at risk of legal consequences).

And I don't know of a way to trigger the issue at will. This happens randomly and sporadically (but it has happened dozens of times in the last few months and keeps happening).

Every time this has happened I have checked at least some of the following, and in a few case all of them:
- the API request by the bot to delete the message (deleteMessage) had not failed, and had received a success response
- any subsequent attempts by the bot to delete the same message "again" resulted in the error saying the message didn't exist, or had already been deleted (I forget the exact phrasing of the API error message)
- the deletion of that message by the bot even appeared in "Recent actions"!

In some cases, the message remains visible to some users and not others. In some cases, it remains visible in Telegram Desktop but not on mobile, even by the same user. Other times, it remains visible both on mobile and Desktop to everybody that we have had a chance to ask to.

I have reported this also to: https://bugs.telegram.org/c/54877 (I'm almost sure I had reported it before and it had been either ignored or even closed). That bug tracker is usually even more useless than this one.

**Comment by 23rd** [comment]
Is your group a megagroup (supergroup)?

**Comment by php4fan** [comment]
They are all supergroups

**Comment by 23rd** [comment]
How often can this happen? Once a day? Once a week? Once a month?

**Comment by Aokromes** [comment]
> Any available public example?

https://t.me/tgbetachat it's quite common.

**Comment by 23rd** [comment]
@php4fan So, I have added a special logging in version 6.2, which may help us catch the problem. Therefore, the next time you come across this problem again, look at log.txt please.

**Comment by Aokromes** [comment]
it's this?

[2025.10.11 20:26:20] History::unknownMessageDeleted. Peer ID: 2377534535, Message ID: 3802.
[2025.10.11 20:26:28] History::unknownMessageDeleted. Peer ID: 1221185967, Message ID: 142792.
[2025.10.11 20:27:22] History::unknownMessageDeleted. Peer ID: 2395080459, Message ID: 5709.
[2025.10.11 20:29:30] History::unknownMessageDeleted. Peer ID: 2407055907, Message ID: 1701.
[2025.10.11 20:55:59] RPC Error: request 22054 got fail with code 400, error PEER_ID_INVALID
[2025.10.11 20:55:59] RPC Error: request 22060 got fail with code 400, error PEER_ID_INVALID
[2025.10.11 20:56:02] RPC Error: request 22066 got fail with code 400, error PEER_ID_INVALID
[2025.10.11 20:56:12] RPC Error: request 22075 got fail with code 400, error PEER_ID_INVALID

<img width="1296" height="1043" alt="Image" src="https://github.com/user-attachments/assets/4ffd0746-1bb2-4f0d-8e96-1c599fd211a4" />

**Comment by 23rd** [comment]
Yes, this is a necessary output, but only a part of it. Well, good starting point.

**Comment by php4fan** [comment]
Still keeps happening...

**Comment by php4fan** [comment]
Issue still exists
